Stripe/StripeiOSTests/ConsumerSessionTests.swift

```diff
@@ -223,18 +164,39 @@ class ConsumerSessionTests: XCTestCase {
             )
 
             let createExpectation = self.expectation(description: "create payment details")
+            let logoutExpectation = self.expectation(description: "logout")
+            let useDetailsAfterLogoutExpectation = self.expectation(description: "try using payment details after logout")
             consumerSession.createPaymentDetails(
                 paymentMethodParams: paymentMethodParams,
                 with: apiClient,
                 consumerAccountPublishableKey: sessionWithKey?.publishableKey
             ) { result in
                 switch result {
                 case .success(let createdPaymentDetails):
-                    if case .card(let cardDetails) = createdPaymentDetails.details {
-                        XCTAssertEqual(cardDetails.expiryMonth, cardParams.expMonth?.intValue)
-                        XCTAssertEqual(cardDetails.expiryYear, cardParams.expYear?.intValue)
-                    } else {
-                        XCTAssert(false)
+                    // If this succeeds, log out...
+                    consumerSession.logout(with: self.apiClient, consumerAccountPublishableKey: sessionWithKey?.publishableKey) { logoutResult in
+                        switch logoutResult {
+                        case .success:
+                            // Try to use the session again, it shouldn't work
+                            consumerSession.createPaymentDetails(paymentMethodParams: paymentMethodParams, with: self.apiClient, consumerAccountPublishableKey: sessionWithKey?.publishableKey) { loggedOutAuthenticatedActionResult in
+                                switch loggedOutAuthenticatedActionResult {
+                                case .success(let success):
+                                    XCTFail("Logout failed to invalidate token")
+                                case .failure(let error):
+
+                                    guard let stripeError = error as? StripeError,
+                                        case let .apiError(stripeAPIError) = stripeError else {
+                                        XCTFail("Received unexpected error response")
+                                        return
+                                    }
+                                    XCTAssertEqual(stripeAPIError.code, "consumer_session_credentials_invalid")
+                                }
+                                useDetailsAfterLogoutExpectation.fulfill()
```

Comment:
Seems like this fulfill should be right under `XCTAssertEqual(stripeAPIError.code`, or we can fulfill prior to asserting.

Reply:
If we fulfill prior to asserting, I think we have a chance of the main thread ending the test as successful before the assertion is checked. I've had to fix that issue in our other tests before.

Reply:
Sorry, i messed up here.. I was thinking something that was just wrong. my bad!